src/xma/src/xmaapi/xmaadmin.cpp

```diff
@@ -247,16 +248,54 @@ xma_admin_session_destroy(XmaAdminSession *session)
     /*
     delete (XmaHwSessionPrivate*)session->base.hw_session.private_do_not_use;
     */
+    XmaHwSessionPrivate *priv1 = (XmaHwSessionPrivate*)session->base.hw_session.private_do_not_use;
```

Comment:
Instead of repeating below code in multiple places, Better to create a destructor and release all the resources in destructor itself